File: api/app/models/attendance.rb
class Attendance < ApplicationRecord
  # Enums
  enum status: { pending: 0, accepted: 1, rejected: 2 }

  # Associations
  belongs_to :member
  belongs_to :session

  # Validations
  validates :status, presence: true, inclusion: { in: statuses.keys }
end

File: client/src/types/Attendance.ts
export type Attendance = {
  id: number;
  session_id: number;
  status: 'pending' | 'accepted' | 'rejected';
  created_at: string;
  updated_at: string;
  member_id: number;
  member_name: string;
  user_id: number;
};
